### PR TITLE
feat: companies sdk

### PR DESCRIPTION
**Pull request by FredPeal**

### CODE REVIEW

File: package-lock.json
Comment (SparoHawk) [inline package-lock.json:1]: **issue(blocking):** Package management is done with Yarn, not NPM.

**suggestion:** Please remove this file and install or update packages with Yarn.

File: src/modules/companies-branches/index.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  }
+
+  public async createCompanyBranch(
+    input: CompanyBranchInput
+  ): Promise<CompanyBranchInterface> {
+    const response = await this.client.mutate({
+      mutation: CREATE_COMPANY_BRANCH,
+      variables: { input },
+    });
+    return response.data.createCompanyBranch;
```
Comment (SparoHawk) [inline src/modules/companies-branches/index.ts:58]: **question(non-blocking):** I see this function's return is not being cast as the expected return type. Is the casting not needed? Did you test to see if this works as intended?

File: src/modules/companies-branches/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  public async updateCompanyBranch(
+    id: string,
+    input: CompanyBranchInput
+  ): Promise<CompanyBranchInterface> {
+    const response = await this.client.mutate({
+      mutation: UPDATE_COMPANY_BRANCH,
+      variables: { id: id, input: input },
+    });
+    return response.data.updateCompanyBranch;
```
Comment (SparoHawk) [inline src/modules/companies-branches/index.ts:69]: **question(non-blocking):** I see this function's return is not being cast as the expected return type. Is the casting not needed? Did you test to see if this works as intended?

File: src/modules/companies/index.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    });
+    return response.data.updateCompany as CompanyInterface;
+  }
+
+  public async deleteCompany(id: string): Promise<Boolean> {
+    const response = await this.client.mutate({
+      mutation: DELETE_COMPANY_MUTATION,
+      variables: { id: id },
+    });
+    return response.data.deleteCompany;
```
Comment (SparoHawk) [inline src/modules/companies/index.ts:83]: **question(non-blocking):** I see this function's return is not being cast as the expected return type. Is the casting not needed? Did you test to see if this works as intended?

File: src/mutations/companies-branches.mutation.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                size
+                field_name
+            }
+            zipcode
+            is_default
+            created_at
+            updated_at
+        }
+    }
+`;
```
Comment (SparoHawk) [inline src/mutations/companies-branches.mutation.ts:55]: **chore(blocking):** Correct code indentation please.

File: src/mutations/companies-branches.mutation.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  mutation addUserToBranch($id: ID!, $user_id: ID!) {
+    addUserToBranch(id: $id, user_id: $user_id)
+  }
+`;
+
+export const REMOVE_USER_FROM_BRANCH = gql`
+    mutation removeUserFromBranch($id: ID!, user_id: ID!){
+        removeUserFromBranch(id: $id, user_id: $user_id)
+    }
+`;
```
Comment (SparoHawk) [inline src/mutations/companies-branches.mutation.ts:73]: **chore(blocking):** Correct code indentation please.

File: src/queries/companies.query.ts
Comment (SparoHawk) [inline src/queries/companies.query.ts:1]: **chore(blocking):** Correct code indentation in this file please.

File: src/types/companies-branches.ts
Comment (SparoHawk) [inline src/types/companies-branches.ts:1]: **question(blocking):** Did you mark all optional properties as such? If not when trying to use these the developer will be required to provide the data.

File: src/types/companies.ts
Comment (SparoHawk) [inline src/types/companies.ts:1]: **question(blocking):** Did you mark all optional properties as such? If not when trying to use these the developer will be required to provide the data.

**Review by SparoHawk — CHANGES_REQUESTED** [review]
**suggestion:** You should define paths for both `mutations` and `queries` in the `tsconfig.json` file in order to replace relative imports for more absolute ones. For example use `@mutations` instead of `../../mutations`.

**Comment by FredPeal** [comment]
> **suggestion:** You should define paths for both `mutations` and `queries` in the `tsconfig.json` file in order to replace relative imports for more absolute ones. For example use `@mutations` instead of `../../mutations`.

That's not working for me. VS Code recognize the import but when I compiled the project, the terminal said "module `mutations` not found, I need help x)"

**Review by SparoHawk — APPROVED** [review]

**Pull request merged**